connectivity/FEATURE_BLE/source/generic/SecurityDb.h

```diff
@@ -181,10 +181,21 @@ class SecurityDb {
             entry_handle_t correct_handle = find_entry_by_peer_ediv_rand(ediv, rand);
             if (!correct_handle) {
                 cb(*db_handle, NULL);
+                return;
             }
             // Note: keys should never be null as a matching entry has been retrieved
             SecurityEntryKeys_t* keys = read_in_entry_local_keys(correct_handle);
             MBED_ASSERT(keys);
+
+            /* set flags connected */
+            SecurityDistributionFlags_t* flags = get_distribution_flags(correct_handle);
+            flags->connected = true;
+
+            /* update peer address */
+            SecurityDistributionFlags_t* old_flags = get_distribution_flags(*db_handle);
```

Comment:
Does it matter to update the peer address ?

Reply:
yes, address changed, otherwise we would've found it by address